### PR TITLE
Refactor to use hooks

### PR DESCRIPTION
**Pull request by jeremyphilemon**
- [x] replace rsc implementation with hooks
- [x] support attachments

### CODE REVIEW

**Comment by vercel-team** [comment]
#### 📝 Changed routes:


| [**/login**](https://gemini-chatbot-3-b7epuxwkv.vercel.sh/login) | [**/register**](https://gemini-chatbot-3-b7epuxwkv.vercel.sh/register) |
|:-:|:-:|
|<a href="https://gemini-chatbot-3-b7epuxwkv.vercel.sh/login">Screenshot of /login</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/fQWKfcK9iUWVwVr4WkAnB1sdkMATz8AAEmSF4Hc9hOeeoiWzn5dg6Sy2tHraNB7X7AXXkZoVYaXk7drQ2zO9AA.png">(view full size)</a> |<a href="https://gemini-chatbot-3-b7epuxwkv.vercel.sh/register">Screenshot of /register</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Yh74q1j0CQTiZMOyNB_Dcbm49XRd6QrxGhSgSKRxaS4qIOv5CkBe0Wqwhdwl6_ejlT0XYlRq-E-A3k-cNrjPbQ.png">(view full size)</a> |



| [**/\[id\]**](https://gemini-chatbot-3-b7epuxwkv.vercel.sh/[id]) | [**/**](https://gemini-chatbot-3-b7epuxwkv.vercel.sh/) |
|:-:|:-:|
|<a href="https://gemini-chatbot-3-b7epuxwkv.vercel.sh/[id]">Screenshot of /[id]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/jyHtiiPHB9xVs-5dhXadqxvYNTJrIdc5LKHdB35CrEp3oPvhg352qFTXJ8QUeM18wjM6M9WxSd38zqsum7PokA.png">(view full size)</a> |<a href="https://gemini-chatbot-3-b7epuxwkv.vercel.sh/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/SWk4tcb8klh2uKLbKCQRL1A48ZEt23T941YlyaLfYbcXuBE_E7npZeNMzwW5AY4ziHh_iFXbHXYnKV3Gy2dOLA.png">(view full size)</a> |





7 deleted routes:
- **/chat/[id]**
- **/**
- **/waiting-room**
- **/login**
- **/new**
- **/share/[id]**
- **/signup**

Commit 9ead41e6b7bab5c409034642219de14b4340b94a (https://gemini-chatbot-3-b7epuxwkv.vercel.sh).

File: README.md [hunk 1]
```diff
@@ -1,49 +1,53 @@
...
+- Data Persistence
+  - [Vercel Postgres powered by Neon](https://vercel.com/storage/postgres) for saving chat history and user data
+  - [Vercel Blob](https://vercel.com/storage/blob) for efficient object storage
+- [NextAuth.js](https://github.com/nextauthjs/next-auth)
+  - Simple and secure authentication
 
 ## Model Providers
 
-This template ships with Google Gemini `models/gemini-1.0-pro-001` as the default. However, thanks to the [Vercel AI SDK](https://sdk.vercel.ai/docs), you can switch LLM providers to [OpenAI](https://openai.com), [Anthropic](https://anthropic.com), [Cohere](https://cohere.com/), [Hugging Face](https://huggingface.co), or using [LangChain](https://js.langchain.com) with just a few lines of code.
+This template ships with Google Gemini `gemini-1.5-pro` models as the default. However, with the [AI SDK](https://sdk.vercel.ai/docs), you can switch LLM providers to [OpenAI](https://openai.com), [Anthropic](https://anthropic.com), [Cohere](https://cohere.com/), and many more with just a few lines of code.
```
Comment (nicoalbanese) [inline README.md:38]: link many more to providers page? sdk.vercel.ai/docs/providers

File: README.md [hunk 2]
```diff
@@ -1,49 +1,53 @@
...
 
 ## Running locally
 
 You will need to use the environment variables [defined in `.env.example`](.env.example) to run Next.js AI Chatbot. It's recommended you use [Vercel Environment Variables](https://vercel.com/docs/projects/environment-variables) for this, but a `.env` file is all that is necessary.
 
-> Note: You should not commit your `.env` file or it will expose secrets that will allow others to control access to your various Google Cloud and authentication provider accounts.
+> Note: You should not commit your `.env` file or it will expose secrets that will allow others to control access to your various OpenAI and authentication provider accounts.
```
Comment (nicoalbanese) [inline README.md:50]: keep this as google? 
should we add .env to the gitignore if it's not already there?

File: app/(chat)/api/chat/route.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+          flightNumber: z.string().describe("Flight number"),
+        }),
+        execute: async ({ flightNumber }) => {
+          const seats = await generateSampleSeatSelection({ flightNumber });
+          return seats;
+        },
+      },
+      createReservation: {
+        description: "Display pending reservation details",
+        parameters: z.object({
```
Comment (nicoalbanese) [inline app/(chat)/api/chat/route.ts:114]: thoughts on defining these schemas once and then reusing across ai/actions.ts and here? could also use the infered types for the component?

File: components/custom/chat.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  id: string;
+  initialMessages: Array<Message>;
+}) {
+  const { messages, handleSubmit, input, setInput, append, isLoading, stop } =
+    useChat({
+      body: { id },
+      initialMessages,
+      maxSteps: 10,
+      onFinish: () => {
+        window.history.replaceState({}, "", `/chat/${id}`);
```
Comment (nicoalbanese) [inline components/custom/chat.tsx:26]: any reasoun to use window instead of useRouter here?

File: components/flights/authorize-payment.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    `/api/reservation?id=${intent.reservationId}`,
+    fetcher,
+  );
+
+  const [input, setInput] = useState("");
+
+  const handleAuthorize = async (magicWord: string) => {
+    try {
+      const response = await fetch(
+        `/api/reservation?id=${intent.reservationId}`,
```
Comment (nicoalbanese) [inline components/flights/authorize-payment.tsx:28]: any reason not using the mutate fn here rather than fetch?

File: package.json [hunk 6]
```diff
@@ -1,80 +1,70 @@
 {
+  "name": "gemini-chatbot",
+  "version": "0.1.0",
   "private": true,
   "scripts": {
     "dev": "next dev",
```
Comment (nicoalbanese) [inline package.json:6]: can we add turbo flag?

File: package.json [hunk 7]
```diff
@@ -1,80 +1,70 @@
...
-    "@upstash/ratelimit": "^1.0.3",
-    "@vercel/analytics": "^1.1.2",
-    "@vercel/kv": "^1.0.1",
-    "@vercel/og": "^0.6.2",
-    "ai": "^3.2.10",
+    "@radix-ui/react-label": "^2.1.0",
+    "@radix-ui/react-slot": "^1.1.0",
+    "@radix-ui/react-tooltip": "^1.1.2",
+    "@radix-ui/react-visually-hidden": "^1.1.0",
+    "@supabase/ssr": "^0.5.1",
```
Comment (nicoalbanese) [inline package.json:21]: remove supabase?

**Review by nicoalbanese — APPROVED** [review]

File: components/custom/chat.tsx (hunk 4 above)
Comment (jeremyphilemon) [reply components/custom/chat.tsx:26]: the intention was to do a shallow route push, and not reload the page

**Pull request merged**